File: GreenDiamond/GreenDiamond/GameUtils.cpp
#include "all.h"

i2D_t PositionToMapCellPoint(double x, double y)
{
	int mapTileX = (int)(x / MAP_TILE_WH);
	int mapTileY = (int)(y / MAP_TILE_WH);

	return makeI2D(mapTileX, mapTileY);
}

File: GreenDiamond/GreenDiamond/Main01/GameUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Charlotte.Tools;

namespace Charlotte.Main01
{
	public static class GameUtils
	{
		public static I2Point PointToMapCellPoint(double x, double y)
		{
			int mapTileX = (int)Math.Floor(x / Consts.MAP_TILE_WH);
			int mapTileY = (int)Math.Floor(y / Consts.MAP_TILE_WH);

			return new I2Point(mapTileX, mapTileY);
		}
	}
}
